src/main/java/spleetwaise/transaction/logic/commands/ClearCommand.java

```diff
@@ -0,0 +1,24 @@
+package spleetwaise.transaction.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import spleetwaise.address.logic.commands.CommandResult;
+import spleetwaise.transaction.model.Model;
+import spleetwaise.transaction.model.TransactionBook;
+
+/**
+ * Represents a command to clear the transaction book.
+ */
+public class ClearCommand extends Command {
+
+    public static final String COMMAND_WORD = "clearTxn";
+    public static final String MESSAGE_SUCCESS = "Transaction book has been cleared!";
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.setTransactionBook(new TransactionBook());
```

Comment:
I feel like this single call reaches too far beyond the current level of abstraction. I suggest we do:

```java
model.clearTransactionBook(); // Implement this
// Then in TransactionBook.java
transactionBook.clear(); // Implement this
```

Reply:
Sure we can explore better implementation for this in future iterations. The current implementation takes the form from `java/spleetwaise/address/logic/commands/ClearCommand.java`

```
public class ClearCommand extends Command {

    public static final String COMMAND_WORD = "clear";
    public static final String MESSAGE_SUCCESS = "Address book has been cleared!";


    @Override
    public CommandResult execute(Model model) {
        requireNonNull(model);
        model.setAddressBook(new AddressBook());
        return new CommandResult(MESSAGE_SUCCESS);
    }
}
```